### PR TITLE
Feat/enhance mergeCell functionality to support more detailed conditions

### PR DESCRIPTION
**Pull request by xielin9527**
[[中文版模板 / Chinese template](https://github.com/VisActor/VTable/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Chore
- [ ] Release
- [ ] Other (about what?)

### 🔗 Related issue link

related #2907
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: docs/assets/option/zh/column/text-column-type.md
Comment (Rui-Sun) [inline docs/assets/option/zh/column/text-column-type.md:1]: 对应的英文文档也同步更新一下吧

File: common/changes/@visactor/vtable/feat-merge-cell_2024-11-28-04-28.json
Comment (Rui-Sun) [inline common/changes/@visactor/vtable/feat-merge-cell_2024-11-28-04-28.json:1]: 详细说明一下具体更新的地方吧

Comment (xielin9527) [reply common/changes/@visactor/vtable/feat-merge-cell_2024-11-28-04-28.json:1]: 已更新

Comment (Rui-Sun) [inline common/changes/@visactor/vtable/feat-merge-cell_2024-11-28-04-28.json:1]: 这个文件删掉吧

File: common/changes/@visactor/vtable/feat-merge-cell_2024-12-02-03-04.json
Comment (Rui-Sun) [inline common/changes/@visactor/vtable/feat-merge-cell_2024-12-02-03-04.json:1]: 这个comment使用英文，格式通git commit

Comment (xielin9527) [reply common/changes/@visactor/vtable/feat-merge-cell_2024-12-02-03-04.json:1]: 能用ts不，json写的有点费劲

**Comment by codeboy0311** [comment]
这个问题这周会修复吗？

File: common/changes/@visactor/vtable/feat-merge-cell_2024-12-02-03-04.json
Comment (xielin9527) [reply common/changes/@visactor/vtable/feat-merge-cell_2024-12-02-03-04.json:1]: done

File: common/changes/@visactor/vtable/feat-merge-cell_2024-12-02-03-04.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "Added the current cell/compared cell location and table instance parameters to column.text.mergeCell when invoked as a function, making them accessible within the function: vtable/src/layout/cell-range/simple-cell-range.ts\nUpdated the type definition of mergeCell: vtable/src/ts-types/common.ts\nUpdated both English and Chinese documentation: column/text-column-type.md\nAdded a new demo document: list/list-merge-cell-function.md",
```
Comment (Rui-Sun) [inline common/changes/@visactor/vtable/feat-merge-cell_2024-12-02-03-04.json:4]: 这里改成：
"feat: add mergeCell function to support more detailed conditions"

**Comment by fangsmile** [comment]
https://visactor.io/vtable/guide/Contribution_Guide  后续提pr可以按照这个文档中提到的rush change-all  生成变更日志哈 @xielin9527

**Review by fangsmile — APPROVED** [review]

**Comment by fangsmile** [comment]
永久有效
请加入飞书群 联系我们，领取开源共建的奖品  @xielin9527

**Pull request merged**